docs/contributing/testing.rst

```diff
@@ -0,0 +1,239 @@
+.. _testing:
+
+=======
+Testing
+=======
+
+Many tests require sample graphs that are too big to keep in this repository.
+You can get them with ``make input``.
+
+If you need to update the inputs, they are referenced as
+https://katana-ci-public.s3.us-east-1.amazonaws.com/inputs/katana-inputs-vN.tar.gz
+in ``.github/workflows``, ``inputs/CMakeLists.txt`` and
+``external/katana/python/katana/exmaple_utils.py``.  ``vN`` is a monotonically
+increasing version number. You can use a command ``inputs/update_inputs.sh`` to
+create create a new input collection. After creating the tar file, you will
+need to upload the file to the public S3 bucket.
```

Comment:
public just means globally accessible here. An external contributor won't be able update this directly.

Reply:
Can you please add that in the docs because the above text talks about internal and external contributors?

Reply:
I don't understand. There is no mention of internal and external contributors in the `testing.rst` document.

I also don't see the point of saying "upload the file here; if you don't have authorization to upload, you won't be able to upload"

Reply:
But it was mentioned in some other document. 

As this is an open repo, it would be good to be explicit if some statements only apply to internal contributors, especially because "public" can be misleading.

Reply:
I actually don't think it is good to draw this distinction. "contributor" is only used two times in the existing documentation, and it is reference to how PRs should be merged, which is a situation where you naturally have contributors with write access to this repo and those who do not. And there is a question about the convention on how to merge changes for people who do have write access.

The people who have write access to some random S3 bucket is not related to (write) repo access at all. And again, it will be very obvious that you don't have access to this bucket because you cannot write to it.

Reply:
Ok, that makes sense.